ert/data/_record.py

```diff
@@ -110,20 +131,34 @@ def record_type(self) -> RecordType:
             f"Not able to deduce record type from data was: {type(self.data)}"
         )
 
-    @root_validator(skip_on_failure=True)
-    def ensure_consistent_index(cls, record: Dict[str, Any]) -> Dict[str, Any]:
-        assert (
-            "data" in record and "index" in record
-        ), "both data and index must be defined for a record"
-        norm_record_index = _build_record_index(record["data"])
-        assert (
-            norm_record_index == record["index"]
-        ), f"inconsistent index {norm_record_index} vs {record['index']}"
-        return record
+
+class BlobRecord(Record, _DataElement):
+    data: blob_record_data
```

Comment:
The index was introduced as a generic way of getting the keys of a record. With that in mind we should perhaps keep an index in the blob record as well?

Reply:
But does an index have a meaning for a blob of unknown data? Even so, we do not use it now, and maybe we do not know yet how it should look like. It could be easily introduced later as needed?

Reply:
Well, the blob data is inherently indexed as long as we store it in a list 🤷🏻 It could even be an issue to just have a blob record store a single blob... If we keep it in a list as is, then I think the index makes sense to keep.

Reply:
I was wondering about the list, but decided not to touch that for now. If it is ok with you, I will assume that this still needs discussion in an upcoming issue, and neither remove the list, nor add an index in this PR.